control-plane-operator/controllers/hostedcontrolplane/v2/karpenteroperator/secret.go

```diff
@@ -0,0 +1,25 @@
+package karpenteroperator
+
+import (
+	"fmt"
+
+	component "github.com/openshift/hypershift/support/controlplane-component"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func adaptCredentialsSecret(cpContext component.WorkloadContext, secret *corev1.Secret) error {
+	hcp := cpContext.HCP
+
+	awsCredentialsTemplate := `[default]
+	role_arn = %s
+	web_identity_token_file = /var/run/secrets/openshift/serviceaccount/token
+	sts_regional_endpoints = regional
+`
```

Comment:
this template could be moved to the secret yaml file. The adapt function can then read the template from the secret object, replace the role_arn field and write it back.

Reply:
you are right, I was thinking of configMaps where we have the template in the yaml file. True, this doesn't work for secrets.